api/applications/applicationRouter.js

```diff
@@ -232,59 +234,124 @@ router.post(
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              items:
+ *                status: 422
+ *                message:  "unexpected value; permitted: 'approved', 'rejected', 'pending'"
+ *       '409':
+ *         description: Response from error on second call involving Auth0 with some failure related to creating / provisioning a new user
+ *         note: many different error responses are possible with many different status codes, the most common one is shown below
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               items:
+ *                 status: 409
+ *                 message:  'User already exists.'
+ *      '422':
+ *        description: Response from error on third operation involving database operation with some failure related to Postgres where insertion of new profile failed
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              items:
+ *                status: 422
+ *                message:  'insert into \"profiles\" (\"profile_id\", \"role\") values ($1, $2) returning * - null value in column \"user_id\" of relation \"profiles\" violates not-null constraint'
  */
-/*
-NOTE: FE discerns if the user is a mentor/mentee and sends back the appropriate shape: 
-mentor = {validate_status & current_company }
-mentee = {validate_status}
-this endpoint can then send the validate status to the appropriate endpoint depending on whether the current_company is present or not
-  
-authRequired, and AdminRequired imports have been left (commented out) because they will likely be needed when auth0 is implemented.
-*/
-router.post(
-  '/update-validate_status/:profile_id',
-  validateStatusRequest,
-  (req, res, next) => {
-    const isMentor = req.body.current_company;
-    const { profile_id } = req.params;
 
-    const role = isMentor ? 'mentor' : 'mentee';
+router.post('/update-validate_status/:profile_id', async (req, res, next) => {
+  const { role, email, status } = req.body;
+  const { profile_id } = req.params;
+  let user_id;
 
-    axios
-      .post(`${baseURL}/update/${role}/${profile_id}`, {
-        validate_status: req.body.validate_status,
-      })
-      .then((result) => {
-        res.send({ status: result.status, message: result.data });
-      })
-      .catch((err) => {
-        next({ status: err.status, message: err.message });
-      });
+  try {
+    await axios.post(`${baseURL}/update/${role}/${profile_id}`, {
+      validate_status: status,
+    });
+    if (status === 'rejected') {
+      res.json({ status: 200, message: `${role} rejected!` });
+      return;
+    }
+  } catch (err) {
+    next({
+      status: err.response.status,
+      message: err.response.data.detail[0].msg,
+    });
+    return;
   }
-);
+
+  try {
+    const payload = {
+      email,
+      password: passGenerator(8),
```

Comment:
8 is bare minimum for our PW constraints, do we want to add a few chars to this?